e2e/e2e_test.go

```diff
@@ -0,0 +1,168 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/utils/pointer"
+)
+
+const (
+	blockingPodName = "api-blocker-pod"
+	safeToAssumeNodeRebootTimeout = 180 * time.Second
+	testNamespace = "default"
+)
+
+var _ = Describe("e2e", func() {
+	var targetNode string
+
+	BeforeEach(func() {
+		// randomly pick a host (or let the scheduler do it by running the blocking pod)
+		// block the api port to make it go Ready Unknown
+		node, err := makeNodeUnready(time.Minute * 5)
+		Expect(err).NotTo(HaveOccurred())
+		targetNode = node
+	})
+
+	AfterEach(func() {
+		removeAPIBlockingPod()
+	})
+	When("Node conditions meets the unhealthy criteria", func() {
+		It("Remediates a host", func() {
+			Expect(nhcRemediatesTheHost(targetNode, 2*time.Minute)).To(Succeed())
+			Expect(nodeObjectIsRemoved(targetNode, safeToAssumeNodeRebootTimeout+30*time.Second)).To(Succeed())
+		})
+	})
+})
+
+func nodeObjectIsRemoved(nodeName string, timeout time.Duration) error {
```

Comment:
hmmm poison pill removes the node but re-creates it right away, so thig might wrongfully fail.
in pp tests, I think we use the node's `CreationTime` to know if it's a new restored node or not.

Reply:
see this:
https://github.com/medik8s/poison-pill/blob/master/controllers/controller_test.go#L130-L138

Reply:
ack